File: day1/part1/parse.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>


int main(){
	uint64_t sum = 0;
	uint64_t sum_max = 0;
	FILE *file = fopen("input.txt", "r");
	uint64_t num;
	char line[100];

	
	while(fgets(line, 100, file)){
		if (line[0] == '\n'){
			if (sum > sum_max) sum_max = sum;
			sum = 0;
		}
		else{
			sscanf(line, "%li\n", &num);
			sum += num;
		}
	}
	printf("Maximum: %li", sum_max);
		
	
}

File: day1/part1/parse.py
#!/bin/python3

name = "input.txt"

def main():
    sum = 0
    sums = []
    print("Start")
    with open(name, "r") as file:
        for line in file:
            # This gets you on the naughty list, not nice to rely on exceptions like this
            try:
                num = int(line)
                sum += num
            except:
                sums.append(sum)
                sum = 0
    print("The sum array is: ")
    print(sums)
    print("The maximum of the array is: ")
    print(max(sums))


if __name__ == "__main__":
    main()
